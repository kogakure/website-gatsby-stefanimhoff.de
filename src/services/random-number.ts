export const getRandomNumber = (max: number) =>
  String(Math.floor(Math.random() * Math.floor(max)));
