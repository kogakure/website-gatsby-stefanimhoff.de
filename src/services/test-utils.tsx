import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
