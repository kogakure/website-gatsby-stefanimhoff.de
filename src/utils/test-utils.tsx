import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';
import { ThemeType } from '../typings/Theme';
import { lightTheme } from '../themes';

interface CustomTheme {
  theme: ThemeType;
}

type RenderOptionsWithTheme = RenderOptions & CustomTheme;

const customRender = (
  node: React.ReactElement,
  options: RenderOptionsWithTheme = { theme: lightTheme }
): RenderResult => {
  const { theme } = options;
  const rendered = render(
    <ThemeProvider theme={theme}>
      <EmojifyProvider>
        <LocalizedDateProvider>{node}</LocalizedDateProvider>
      </EmojifyProvider>
    </ThemeProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (
      ui: React.ReactElement,
      rerenderOptions?: RenderOptionsWithTheme
    ) => customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
