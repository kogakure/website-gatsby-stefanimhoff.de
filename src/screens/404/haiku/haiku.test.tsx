import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../../services/test-utils';

import { Haiku } from '.';

describe('404', () => {
  describe('Haiku', () => {
    test('renders correctly', () => {
      const { container } = render(<Haiku />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(<Haiku />);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });
});
