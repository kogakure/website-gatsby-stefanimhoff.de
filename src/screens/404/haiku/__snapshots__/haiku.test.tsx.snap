// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Haiku renders correctly 1`] = `
.c0 {
  margin-bottom: clamp(1.44rem,3.70vw,1.728rem);
  margin-left: 0;
  margin-right: 0;
  margin-top: clamp(1.44rem,3.70vw,1.728rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-style: italic;
}

.c0 p {
  margin: 0;
}

.c2 {
  font-family: inherit;
  margin-bottom: 0;
  margin-top: 0;
}

.c1 {
  grid-column: 1 / -1;
  grid-row: 2;
  z-index: 1;
}

@media (min-width:768px) {
  .c1 {
    grid-column: 3 / -6;
  }
}

<blockquote
  class="c0 c1"
>
  <pre
    class="c2"
  >
    You searched for something.
    <br />
    It might have once been here.
    <br />
    But now itâ€™s gone.
  </pre>
</blockquote>
`;
